armi/cases/case.py

```diff
@@ -196,10 +197,28 @@ def dependencies(self):
                 )
             )
         # ensure that a case doesn't appear to be its own dependency
+        dependencies.update(self._dependencies)
         dependencies.discard(self)
 
         return dependencies
 
+    def addExplicitDependency(self, case):
+        """
+        Register an explicit dependency
+
+        When evaluating the ``dependency`` property, dynamic dependencies are probed
+        using the current case settings and plugin hooks. Sometimes, it is necessary to
+        impose dependencies that are not expressed through settings and hooks. This
+        method stores another case as an explicit dependency, which will be included
+        with the other, implicitly discovered, dependencies.
+        """
+        if case in self._dependencies:
+            runLog.warn(
+                "The case {} is already explicity specified as a dependency of "
+                "{}".format(case, self)
+            )
```

Comment:
Is a warning necessary here if we are using a set for `self._dependencies`? Also, is this supposed to be `runLog.warn` or `runLog.warning`? I don't have the code in front of me so just checking.

Reply:
I think it's better than not having it. Re-adding a dependency that's already there is a pretty good sign that the caller is not 100% sure what they're doing. Notice that this does not produce a warning if there is overlap between what's stored in `_dependencies` and what cases get discovered through evaluation of the `dependencies` property.

Good catch on `warn` fixed it.